core-common/src/main/java/org/zalando/nakadi/repository/kafka/KafkaLocationManager.java

```diff
@@ -35,11 +36,31 @@ public KafkaLocationManager(final ZooKeeperHolder zkFactory, final KafkaSettings
         this.zkFactory = zkFactory;
         this.kafkaProperties = new Properties();
         this.kafkaSettings = kafkaSettings;
+        applySecurityProperties();
         this.updateBootstrapServers(true);
         this.scheduledExecutor = Executors.newSingleThreadScheduledExecutor();
         this.scheduledExecutor.scheduleAtFixedRate(() -> updateBootstrapServersSafe(false), 1, 1, TimeUnit.MINUTES);
     }
 
+    private void applySecurityProperties() {
+        if (this.kafkaSettings.getSecurityProtocol().isPresent()) {
+            this.kafkaProperties.put(CommonClientConfigs.SECURITY_PROTOCOL_CONFIG,
+                    this.kafkaSettings.getSecurityProtocol().get());
+        }
+        if (this.kafkaSettings.getSaslMechanism().isPresent()) {
```

Comment:
I assume the configuration should be applied as a whole (protocol, mechanism, users, passwords) or nothing should be applied. otherwise it will be in inconsistent state.

Reply:
I like the idea that uncompleted security configuration wont break Nakadi, and in this case I will be verbal in logs to explain that a missing piece of configuration wont enable security to instruct the engineer that some part of configuration is missing.